### PR TITLE
feat: support generating disk images with image cache

### PR DESCRIPTION
**Pull request by frezbo**
Add support for generating disk images with image cache.

Fixes: #9616

### CODE REVIEW

**Comment by frezbo** [comment]
One thing i can think of is if the image cache is too large, the disk image creation would fail since we set a default sparse file of almost 1.3gig

```yaml
imageOptions:
    diskSize: 1306525696
```

File: internal/pkg/partition/constants.go [hunk 1]
```diff
@@ -25,6 +25,7 @@ const (
 	FilesystemTypeZeroes FileSystemType = "zeroes"
 	FilesystemTypeXFS    FileSystemType = "xfs"
 	FilesystemTypeVFAT   FileSystemType = "vfat"
+	FileSystemTypeExt4   FileSystemType = "ext4"
```
Comment (frezbo) [inline internal/pkg/partition/constants.go:28]: maybe split this into a different PR :thinking:

File: internal/pkg/partition/partition.go [hunk 2]
```diff
@@ -90,6 +90,13 @@ func NewPartitionOptions(label string, uki bool) Options {
 			PartitionType:  LinuxFilesystemData,
 			Size:           0,
 		}
+	case constants.ImageCachePartitionLabel:
+		return Options{
+			FormatOptions:  *formatOptions,
+			PartitionLabel: label,
+			PartitionType:  LinuxFilesystemData,
+			Size:           0,
```
Comment (frezbo) [inline internal/pkg/partition/partition.go:98]: should this be set to the size of the incoming cache?

File: internal/pkg/partition/constants.go (hunk 1 above)
Comment (smira) [reply internal/pkg/partition/constants.go:28]: should be good, it's only used in the installer for now

File: internal/pkg/partition/partition.go (hunk 2 above)
Comment (smira) [reply internal/pkg/partition/partition.go:98]: good question... I guess this makes sense (or specified as a parameter?)

Comment (frezbo) [reply internal/pkg/partition/partition.go:98]: actually it shouldn't matter much here, probably the overall profile disk size should be bumped, another option is to increment the default with the size of cache , so it'll grow automatically

Comment (frezbo) [reply internal/pkg/partition/partition.go:98]: ahh the image already supports setting disk size, we can document to bump it if using huge image cache

**Review by dsseng — APPROVED** [review]

**Review by smira — APPROVED** [review]
should be good to go, tested this locally (missing docs, but we miss them everywhere for the Image Cache)
